File: python_sources/kaggle/house_prices/outliers.py


condition = 'GrLivArea > 4000 and SalePrice < 2e5'


def remove_outliers(dataset):
    return dataset.query("not(%s)" % condition).reset_index(drop=True)


def get_strange_cases(dataset):
    return dataset.query(condition).reset_index(drop=True)

File: R_sources/house_prices/outliers.R

within(list(), 
{
    condition <- quote(GrLivArea > 4000 & !is.na(SalePrice) & SalePrice < 2e5)
    
    remove_outliers <- function(dataset) {
        dataset %>% 
        filter( !(!!condition) )
    }
    
    get_strange_cases <- function(dataset) {
        dataset %>%
        filter(!!condition)
    }
     
})
